brooklyn-server/utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java

```diff
@@ -18,16 +18,24 @@
  */
 package org.apache.brooklyn.util.text;
 
+import java.security.SecureRandom;
 import java.util.Random;
 
+import com.google.common.base.Preconditions;
+
 public class Identifiers {
     
     private static Random random = new Random();
 
-    /** @see #JAVA_GOOD_PACKAGE_OR_CLASS_REGEX */ 
-    public static final String JAVA_GOOD_START_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_";
-    /** @see #JAVA_GOOD_PACKAGE_OR_CLASS_REGEX */ 
-    public static final String JAVA_GOOD_NONSTART_CHARS = JAVA_GOOD_START_CHARS+"1234567890";
+    public static final String UPPER_CASE_ALPHA = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    public static final String LOWER_CASE_ALPHA = "abcdefghijklmnopqrstuvwxyz";
+    public static final String NUMERIC = "1234567890";
+    public static final String NON_ALPHA_NUMERIC = "!$%@#";
```

Comment:
I'd prefer to add more but went with the default set used by [lastpass](https://lastpass.com/generatepassword.php) in case others were problematic - possibly this isn't a problem anymore.